### PR TITLE
Feature: support listing of descriptors and read/write operations on them

### LINKED ISSUE #322: Support for Descriptors
For most of the BLE devices, we need to write/read/notify from their descriptors. As far as I checked, this library does not support this functionality. Is there any work around or plans to support descriptors?

### PR DESCRIPTION
**Pull request by mikolak**
Resolves #322

### CODE REVIEW

File: lib/internal/constants.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (dariuszseweryn) [inline lib/internal/constants.dart]: Needed?

File: example/lib/test_scenarios/peripheral_test_operations.dart [hunk 2]
```diff
@@ -40,28 +40,64 @@ class PeripheralTestOperations {
...
-      List<Service> services = await peripheral.services();
-      log("PRINTING SERVICES for \n${peripheral.name}");
-      services.forEach((service) => log("Found service \n${service.uuid}"));
-      Service service = services.first;
-      log("PRINTING CHARACTERISTICS FOR SERVICE \n${service.uuid}");
-
-      List<Characteristic> characteristics = await service.characteristics();
-      characteristics.forEach((characteristic) {
-        log("${characteristic.uuid}");
+  Future<void> discovery() async => await _runWithErrorHandling(() async {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply example/lib/test_scenarios/peripheral_test_operations.dart]: It is only for manual supervised test. All the function `_runWithErrorHandling()` does is print errors to UI and logging service.

File: android/src/main/java/com/polidea/flutter_ble_lib/converter/CharacteristicsResponseJsonConverter.java [hunk 3]
```diff
@@ -1,30 +0,0 @@
-package com.polidea.flutter_ble_lib.converter;
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/flutter_ble_lib/converter/CharacteristicsResponseJsonConverter.java]: Yeah, unused class.

File: android/src/main/java/com/polidea/flutter_ble_lib/converter/MultiDescriptorsResponseJsonConverter.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        String CHARACTERISTIC_ID = "id";
+        String DESCRIPTORS = "descriptors";
+    }
+
+    @Override
+    public String toJson(List<Descriptor> descriptors) throws JSONException {
+        JSONObject root = new JSONObject();
+        JSONArray array = new JSONArray();
+        if (descriptors.size() > 0) {
+            root.put(Metadata.SERVICE_ID, descriptors.get(0).getServiceId());
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/flutter_ble_lib/converter/MultiDescriptorsResponseJsonConverter.java]: It is used only for descriptors from a single characteristic.

File: example/lib/test_scenarios/peripheral_test_operations.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (dariuszseweryn) [reply example/lib/test_scenarios/peripheral_test_operations.dart]: There could be an explanation somewhere about it I suppose

**Pull request merged**